lessons/ro/chapter_7.yaml

```diff
@@ -0,0 +1,272 @@
+
+    * Structurile nu pot mosteni campuri de la o structura parinte.
+
+    * Structurile nu pot mosteni functii de la o structura parinte.
+
+    Acestea fiind spuse, Rust implementeaza o multime de caracteristici ale unui limbaj de programare,
+    deci probabail nu va vor deranja aceste lipsuri.
+- title: Incapsulare prin matode
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cfca759d42182509a46f65236bdfbf61
+  content_markdown: >
+    Rust suporta conceptul de *obiect* ce reprezinta o structura asociata unor functii (cunoscute si ca *metode*).
+
+    Primul parametru al oricarei metode trebuie sa fie o referinta la instanta asociata apelului de metoda
+    (exemplu: `instantaObiectului.foo()`). Rust foloseste:
+
+    * `&self` - Referinta imutabila la instanta.
+
+    * `&mut self` - Referinta mutabila la instanta.
+
+    Metodele sunt definite in cadrul unui bloc de implementare prin cuvantul cheie `impl`:
+
+    ```rust
+
+    impl StructuraMea { 
+        ...
+        fn foo(&self) {
+            ...
+        }
+    }
+
+    ```
+- title: Abstractie prin Expunere Selectiva
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=859b684d23c45987c00ab1ad47584193
+  content_markdown: >
+    Rust poate ascude implementarile interioare ale obiectelor.
+
+    In mod implicit, campurile si metodele sunt accesibile doar modulului de care apartin.
+
+    Cuvantul cheie `pub` expune campurile si metodele structurii in afara modulului.
+- title: Polimorfismul prin Trasaturi
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=93c3841112810ceb5014298d1ee12ac7
+  content_markdown: >
+    Rust suporta polimorfismul prin trasaturi (traits). Trasaturile ne permit associerea unui set
+    de metode cu o structura.
+
+    Prima data definim semnaturile metodelor unei trasaturi in:
+
+     ```
+
+    trait TrasaturaMea {
+        fn foo(&self);
+        ...
+    }
+
+    ```
+    Cand o structura implementeaza o trasatura, este stabilit un contract ce permite interactiunea noastra 
+    indirecta cu structura, prin tipul trasaturii (exemplu: `&dyn TrasaturaMea` ), fara a fi nevoie
+    sa cunoastem tipul real al structurii.
+
+    Metodele trasaturii implementate de structura sunt definite in cadrul unui bloc de implementare:
+
+    ```rust
+
+    impl TrasaturaMea for StructuraMea { 
+        fn foo(&self) {
+            ...
+        }
+        ... 
+    }
+
+    ```
+- title: Metode Implementate Trasaturilor
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e7f8bf360fc76c991ac68ce65de812c9
+  content_markdown: >
+    Trasaturile pot avea metode implementate.
+
+    Functiile nu au acces direct asupra campurilor interioare ale unei structuri, dar acestea pot fi
+    folositoare in distribuirea comportamentelor intre mai multi implementarori ai trasaturilor.
```

Comment:
What does this actually mean?

Reply:
i will try rephrasing it

Reply:
Rephrased it in the last commit. Is it clearer now?